llmclient/llms.py

```diff
@@ -618,19 +619,31 @@ async def acomplete_iter(  # type: ignore[override]
     @rate_limited
     async def achat(self, messages: list[Message]) -> Chunk:  # type: ignore[override]
         prompts = [m.model_dump(by_alias=True) for m in messages if m.content]
-        response = await track_costs(self.router.acompletion)(self.name, prompts)
+        response = await track_costs(self.router.acompletion)(self.name, prompts)  # type: ignore[arg-type]
+        choice = response.choices[0]
+        reasoning_content = None
+        if isinstance(choice, litellm.Choices) and hasattr(
+            choice.message, "provider_specific_fields"
+        ):
+            provider_specific_fields = choice.message.provider_specific_fields
+            if isinstance(provider_specific_fields, dict):
+                reasoning_content = provider_specific_fields.get(
+                    "reasoning_content", None
+                )
```

Comment:
```suggestion
        if isinstance(choice, litellm.Choices) and hasattr(
            choice.message, "provider_specific_fields"
        ) and isinstance(choice.message.provider_specific_fields, dict):
            reasoning_content = choice.message.provider_specific_fields.get(
                "reasoning_content", None
            )
```

We can combine these